### DISCUSSION TITLE
/@vite/client should have a configurable base path

### DISCUSSION THREAD

**Discussion by CharlesStover** [QwikDev/qwik-evolution#22] [Proposals For Qwik]
#### Problems you are trying to solve:
* When mounting my Qwik app behind a reverse proxy, the relative Vite client path `/@vite/client` becomes invalid and 404s.

#### Goals you are trying to achieve:
* I want my Vite client to mount from `localhost`, but I am testing my Vite app on a different custom domain.

#### Any other context or information you want to share:
The Vite client path is [hard coded here](https://github.com/QwikDev/qwik/blob/a86f56e7117e9ac566c756b21dd76acaed3ea306/packages/qwik/src/optimizer/src/plugins/vite-dev-server.ts#L425).

---

#### What do you propose?
If the Qwik vite plugin took a prop like `viteClientBase` (default: `/`), this would be configurable and solve my use case.

#### Code examples

``` 
qwikVite({
  viteClientBase: 'http://localhost:3000/',
});
```

---

### Links / References

https://github.com/QwikDev/qwik/blob/a86f56e7117e9ac566c756b21dd76acaed3ea306/packages/qwik/src/optimizer/src/plugins/vite-dev-server.ts#L425

**Comment by wmertens** [comment]
I think we'll address this by supporting the environment api, in which vite will take over all dev server tasks
